File: 3147-taking-maximum-energy-from-the-mystic-dungeon/3147-taking-maximum-energy-from-the-mystic-dungeon.py
class Solution:
    def maximumEnergy(self, energy: List[int], k: int) -> int:
        maxEnergy, n = -1111, len(energy)
        for p in range(n-1, n-k-1, -1):
            curr = 0
            for i in range(p, -1, -k):
                curr += energy[i]
                maxEnergy = max(maxEnergy, curr)
        return maxEnergy

File: 3147-taking-maximum-energy-from-the-mystic-dungeon/3147-taking-maximum-energy-from-the-mystic-dungeon.java
class Solution {
    public int maximumEnergy(int[] energy, int k) {
        var maxEnergy = -1111;
        var n = energy.length;
        for(var p=n-1; p>=(n-k); p--){
            var curr = 0;
            for(var i=p; i>=0; i-=k){
                curr += energy[i];
                maxEnergy = Math.max(maxEnergy, curr);
            }
        }
        return maxEnergy;
    }
}